pyproject.toml

```diff
@@ -118,6 +118,16 @@ test = [
   "pillow",
   "filelock",
   "httpx",
+
+  # BYOB
+  ## S3
+  "boto3>=1.34.0",
+  "moto[s3]>=5.0.0",
+  ## Azure
+  "azure-storage-blob>=12.24.0",
+  ## GCP
+  "google-cloud-storage>=2.7.0",
```

Comment:
I think you will want extras for the 3 clouds for users to install as well.

Then you can guarantee this will work e.g. like:
```
pip install weave[gcp]
```

Reply:
Users needn't have any of this - it is only for the backend service

Reply:
I still extracted a sublist for clarity